content/en/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md

```diff
@@ -693,6 +699,298 @@ kubectl apply -f my-crontab.yaml
 crontab "my-new-cron-object" created
 ```
 
+## Validation rules
+
+{{< feature-state state="alpha" for_k8s_version="v1.23" >}}
+
+Validation rules are in alpha since 1.23 and validate custom resources when the
+`CustomResourceValidationExpressions` [feature
+gate](/docs/reference/command-line-tools-reference/feature-gates/) enabled and the schema is a
+[structural schema](#specifying-a-structural-schema).
+
+Validation rules use the [Common Expression Language (CEL)](https://github.com/google/cel-spec)
+to validate custom resource values. Validation rules are included in
+CustomResourceDefinition schemas using the `x-kubernetes-validations` extension.
+
+The Rule is scoped to the location of the `x-kubernetes-validations` extension in the schema.
+And `self` variable in the CEL expression is bound to the scoped value.
+
+For example:
+
+```yaml
+    ...
+    openAPIV3Schema:
+      type: object
+      properties:
+        spec:
+          type: object
+            x-kubernetes-validation-rules:
+              - rule: "self.minReplicas <= self.replicas"
+                message: "replicas should be greater than or equal to minReplicas."
+              - rule: "self.replicas <= self.maxReplicas"
+                message: "replicas should be smaller than or equal to maxReplicas."
+          properties:
+            ...
+            minReplicas:
+              type: integer
+            replicas:
+              type: integer
+            maxReplicas:
+              type: integer
```

Comment:
for completeness, I'd recommend the three properties be marked as required, since the expression is not checking for presence

Reply:
Added required constraints.